src/PrestaShopBundle/Service/Mail/MailTemplateFolderCatalog.php

```diff
@@ -0,0 +1,191 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Service\Mail;
+
+use PrestaShop\PrestaShop\Core\Exception\InvalidException;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Finder\SplFileInfo;
+
+/**
+ * This is a basic mail templates catalog, not a lot of intelligence it is based
+ * simply on existing files on the $mailThemesFolder (no database, or config files).
+ */
+class MailTemplateFolderCatalog implements MailTemplateCatalogInterface
```

Comment:
not a huge fan of this (too much responsibilities in one class) but let's say it's ok

Reply:
the idea behind this is that in the future maybe we will store mail themes with config files, or in database objects
this first `MailTemplateCatalogInterface` is simply based on a folder/file convention but there's room for improvement!